apply cosmos sdk v0.47.x

yoosah opened the pull request:
* cometbft

* ibc v7

* cometbft-db

* apply cosmos v0.47.x

* enable build

* enable test

* update go.mod to apply v0.47.9

* fix evmos naming to ethermint

* formatting

<!-- < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < ☺
v                               ✰  Thanks for creating a PR! ✰    
v    Before smashing the submit button please review the checkboxes.
v    If a checkbox is n/a - please still include it but + a little note why
☺ > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > >  -->

Closes: #XXX

## Description

<!-- Add a description of the changes that this PR introduces and the files that
are the most critical to review.
-->

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

yoosah committed:
Merge branch 'private/cosmos_47' into feature/cosmos_sdk_v47

* cometbft

* ibc v7

* cometbft-db

* apply cosmos v0.47.x

* enable build

* enable test

* update go.mod to apply v0.47.9

* fix evmos naming to ethermint

* formatting

nowooj commented on app/ante/authz_test.go:
```diff
@@ -274,7 +274,7 @@ func (suite *AnteTestSuite) TestRejectDeliverMsgsInAuthz() {
 				),
 			},
-			expectedCode: sdkerrors.ErrUnauthorized.ABCICode(),
+			expectedCode: sdkerrors.ErrUnpackAny.ABCICode(),
```
It must be `ErrUnauthorized`. It looks need to initialized `MsgEthereumTx`

nowooj commented on app/ante/authz_test.go:
```diff
@@ -287,7 +287,7 @@ func (suite *AnteTestSuite) TestRejectDeliverMsgsInAuthz() {
 				),
 			},
-			expectedCode: sdkerrors.ErrUnauthorized.ABCICode(),
+			expectedCode: sdkerrors.ErrUnpackAny.ABCICode(),
```
Same to above.

nowooj commented:
> `github.com/cosmos/cosmos-sdk/types.Int` is deprecated. It looks better to use `cosmossdk.io/math.Int`

Checked https://github.com/xpladev/ethermint/pull/6/commits/4d696b2599d4a261665349d9fddad7071f557244

nowooj commented on server/flags/flags.go:
```diff
@@ -86,7 +86,7 @@ const (
 // AddTxFlags adds common flags for commands to post tx
 func AddTxFlags(cmd *cobra.Command) (*cobra.Command, error) {
-	cmd.PersistentFlags().String(flags.FlagChainID, "testnet", "Specify Chain ID for sending Tx")
+	cmd.PersistentFlags().String(flags.FlagChainID, "ethermint_9000-1", "Specify Chain ID for sending Tx")
```
I think it would be good to set the default chainID to an empty.

nowooj replied:
How are you going to do this part?

nowooj commented:
> `buf.gen.gogo.yaml` & `buf.gen.swagger.yaml`'s `github.com/gogo/protobuf` can be changed to `github.com/cosmos/gogoprpto`

Checked https://github.com/xpladev/ethermint/pull/6/commits/f317e0dcb40e69d0ddd12bcf53319cf9ef1c8968

yoosah committed:
fix: normalized the incorrect test

yoosah committed:
fix: use cosmossdk.io/math.Int

use 'cosmossdk.io/math.Int' type instead of
'github.com/cosmos/cosmos-sdk/types.Int' type

yoosah committed:
fix: use github.com/cosmos/gogoproto

yoosah committed:
feat: update go.mod to apply v0.47.10
